src/realm/util/file.hpp

```diff
@@ -31,6 +31,16 @@
 #include <dirent.h> // POSIX.1-2001
 #endif
 
+#if defined(_MSC_VER) && _MSC_VER >= 1900 // compiling with at least Visual Studio 2015
+#include <experimental/filesystem>
+namespace std {
+    namespace filesystem = std::experimental::filesystem::v1;
+}
+#define REALM_HAVE_STD_FILESYSTEM 1
+#else
+#define REALM_HAVE_STD_FILESYSTEM 0
```

Comment:
`std::filesystem` exists in libstdc++ and libc++ as well, no? Or do we only want to use it on Windows? A FIXME for the future might be appropriate.

Reply:
Indeed it does, although I'm not sure how well it's implemented in GCC 4.9, for example. I know for a fact it works on MSVC 14 and up, but I didn't want to speculate and break Unix-y systems.

Reply:
I'll fix the static thing etc (it's also not thread safe)